azkaban-common/src/main/java/azkaban/executor/Status.java

```diff
@@ -30,6 +34,16 @@
   QUEUED(110),
   FAILED_SUCCEEDED(120),
   CANCELLED(130);
+  // status is TINYINT and in H2 DB the possible values are: -128 to 127
+  // so trying to store CANCELLED in H2 fails at the moment
+
+  private static final Map<Integer, Status> numValMap = new HashMap<>();
```

Comment:
I pushed separately here so that you can take a look:
- truly immutable: https://github.com/juhoautio/azkaban/commit/23c9cdd43e6f74ecefcd3e246dad148633261abe
- a bit of cheating but eventually immutable: https://github.com/juhoautio/azkaban/commit/bf0a7ee646270b67d0ba361bfec08dd54b41aa8d

Do you think it's worth adding this much extra code to make it immutable?

It's pretty much closed down like it is any way..